### PR TITLE
Update helpers to search for users unscoped

### PR DESCRIPTION
**Pull request by sephirothkod**
This is to fix error where `User.find_by` fails to find existing guest users when the app has `User.default_scope` set to `where( guest:false )`

many hyku/hyrax/spotlight/blacklight apps have this setting and it causes errors to crop up when the `guest_user` method is called so logged in users will never see this error.

### CODE REVIEW

**Review by orangewolf — APPROVED** [review]
I can confirm that this fix solved the problem in a Hyku app and separately in a Spotlight app when a user logs out.

**Comment by jcoyne** [comment]
Forgive me for being the devils advocate here, but perhaps Hyrax should not use default scope.  Many people have written about why it's a bad idea. https://medium.com/@vatricemir/a-closer-look-at-rails-default-scope-eb260744f196  I'm not sure it's a great idea to impose on unrelated gems to fix the problems of others.

**Comment by orangewolf** [comment]
@jcoyne I think that’s a fair argument to make for sure. We had seen it in Hyrax and in a few blacklight applications that use this gem and had assumed it was a common pattern. 

The article you linked to, which is more well balanced than many on the topic, seems to call out this case though. 

Emir states “ In my opinion, yes, there are cases when separating the data is extremely important, and using unscoped records would be an exception from the rule. We can look at two examples, first, you are building a multitenant, and second, you primarily in 99% of the cases want to show that specific set of data, like in cases when you would be using act_as_paranoid gem”

To me, when using this gem, it seems likely to fall in to that second case. Often there is an admin panel with the users and their roles. Would you want to show the guests there? Probably not. What about in an export or report of active users? Again, skip the guests. You only really want the guests when you are specifically creating or removing them. 

Now, this gem isn’t where the default scope is set, so I can buy an argument that it shouldn’t be cleared here, but I do think wanting to hide the guests by default is the more common use case. I could see setting a `where` to specifically get the guests regardless of if they are excluded in default scope more specifically (instead of just bulk unscoping) . Or we could set it up to be configurable scope lambda which defaults to a noop. 

What are your thoughts?

**Comment by pacso** [comment]
@sephirothkod - Can you update the changelog please?

**Comment by jcoyne** [comment]
@orangewolf I don't know why one would use this gem if you don't want to show these users.  I'm concerned that a default scope could have other implications with this gem (either now or added in the future), especially since the default scope applies to `.build` and `.create`.   We have no tests for this behavior.  Has anyone confirmed that https://github.com/cbeer/devise-guests/blob/master/lib/railties/devise_guests.rake#L9 still works with this default scope?

**Comment by sephirothkod** [comment]
@jcoyne I have updated the rake task to support scoped users. I understand that you don't know why one would use the gem this way, but reality is that many people do. It is less work for many existing apps to support a default scope where guest users are hidden, rather than filtering out guests in the many, many places that users are called. The article that you linked to actually states this (as orangewolf has already pointed out).

As for default scope affecting new user creations, yes it does, however it is easy to circumnavigate by simply defining user.guest = true during the creation. Devise guests already does this [here](https://github.com/cbeer/devise-guests/blob/0a9c0038c5d1d8790907c3bd56cafd0e87525e15/lib/devise-guests/controllers/helpers.rb#L54) so it is already accounted for. We do have to be careful in the future if we change how creation is done, or add another creation within the gem, but it is in the changelog and we have had this large discussion about it now so it shouldn't be an issue.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.8.2] - 2021-12-22
```
Comment (pacso) [inline CHANGELOG.md:7]:
```suggestion
## [Unreleased]
```

File: lib/devise-guests/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module DeviseGuests
-  VERSION = "0.8.1"
+  VERSION = "0.8.2"
```
Comment (pacso) [inline lib/devise-guests/version.rb:2]: The version gets updated during a release, not on this branch.
```suggestion
  VERSION = "0.8.1"
```

**Comment by pacso** [comment]
Whilst this may be a useful addition for some, it is likely to be a breaking change for others. Apps which have scopes on their user models may require them for things to work properly.

Rather than make assumptions on how this should behave for everybody, it would be preferable to make the behaviour configurable, with the default sticking to current behaviour but with the option of enabling the unscoping behaviour if required.

**Comment by pacso** [comment]
Since no further comments or changes have been made, I'll close this for now.

**Pull request closed without merging**